WarshipGame/Assets/Scripts/AI/Enemy.cs

```diff
@@ -150,4 +166,23 @@ private void OnDrawGizmos()
         if(!GizmosOn)return;
         Gizmos.DrawWireSphere(transform.position, (Radius * 0.866f));
     }
+    
+    private void OnCollisionEnter(Collision hexTileCollision)
+    {
+        if (hexTileCollision.gameObject.layer != _waterLayer) return;
+
+        HexData currentHex = hexTileCollision.gameObject.GetComponent<HexData>();
+
+        _lastDetectedHexType = currentHex.HexType;
+        currentHex.HexType = HexType.Occupied;
+    }
+    
+    private void OnCollisionExit(Collision hexTileCollision)
+    {
+        if (hexTileCollision.gameObject.layer != _waterLayer) return;
```

Comment:
_waterLayer can be swaped with 4 as it is already hardcoded as 4

Reply:
facts frfr